gnomad/assessment/sanity_checks.py

```diff
@@ -104,78 +118,86 @@ def make_filters_sanity_check_expr(
-    :param subpop: Subpop abbreviation, supplied only if subpopulations are included in the annotation groups being checked.
+    :param t: Input MatrixTable or Table containing annotations to be summed.
+    :param subset: String indicating sample subset.
+    :param label_groups: Dictionary containing an entry for each label group, where key is the name of the grouping, e.g. "sex" or "pop", and value is a list of all possible values for that grouping (e.g. ["XY", "XX"] or ["afr", "nfe", "amr"]).
     :param sort_order: List containing order to sort label group combinations. Default is SORT_ORDER.
-    :return: None
+    :param delimiter: String to use as delimiter when making group label combinations.
+    :param metric_first_label: If True, metric precedes label group, e.g. AC-afr-male. If False, label group precedes metric, afr-male-AC.
+    :return: Dictionary of sample sum field check expressions and display fields
     """
-    if prefix != "":
-        prefix = f"{prefix}_"
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
 
-    label_combos = make_label_combos(label_groups)
-    combo_AC = [ht.info[f"{prefix}AC_{x}"] for x in label_combos]
-    combo_AN = [ht.info[f"{prefix}AN_{x}"] for x in label_combos]
-    combo_nhomalt = [ht.info[f"{prefix}nhomalt_{x}"] for x in label_combos]
+    # Check if subset is an empty string which is added to check entire callset but does not need the added delimiter
+    if subset:
+        subset += delimiter
 
+    label_combos = make_label_combos(label_groups, label_delimiter=delimiter)
+    # Grab the adj group for checks as we do not retain the raw metric counts for all sample groups so we cannot check the sample sums
     group = label_groups.pop("group")[0]
-    alt_groups = "_".join(
+    # sum_group is the group that when their metrics are added together they should equal the metric total `group` e.g., if group = callset1, one sum_group could be callset1-pop
+    sum_group = delimiter.join(
         sorted(label_groups.keys(), key=lambda x: sort_order.index(x))
     )
+    info_fields = t.info.keys()
 
-    annot_dict = {
-        f"sum_AC_{group}_{alt_groups}": hl.sum(combo_AC),
-        f"sum_AN_{group}_{alt_groups}": hl.sum(combo_AN),
-        f"sum_nhomalt_{group}_{alt_groups}": hl.sum(combo_nhomalt),
-    }
+    # Loop through metrics to build dictionary of fields that make up 'group' and then stores their sum
+    annot_dict = {}
+    for metric in ["AC", "AN", "nhomalt"]:
+        sum_group_exprs = []
+        for label in label_combos:
+            if metric_first_label:
+                field = f"{metric}{delimiter}{subset}{label}"
+            else:
+                field = f"{subset}{metric}{delimiter}{label}"
 
-    ht = ht.annotate(**annot_dict)
+            if field in info_fields:
+                sum_group_exprs.append(t.info[field])
+            else:
+                logger.info("%s is not in table's info field", field)
```

Comment:
I get these in the output, I think this would have identified that we were missing whole groups in our dataset, thank you so much for adding!!
```
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-papuan-adj is not in table's info field.   !!!! Is this OK, or not? I also think we are missing some of the needed subsets to check here if comparing to what we used to have 
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-biakapygmy-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-san-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-mbutipygmy-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-melanesian-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-papuan-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-biakapygmy-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-san-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-mbutipygmy-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-melanesian-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-papuan-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-biakapygmy-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-san-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-mbutipygmy-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-melanesian-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-papuan-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-papuan-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-biakapygmy-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-biakapygmy-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-san-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-san-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-mbutipygmy-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-mbutipygmy-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-melanesian-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AC-hgdp-melanesian-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-papuan-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-papuan-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-biakapygmy-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-biakapygmy-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-san-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-san-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-mbutipygmy-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-mbutipygmy-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-melanesian-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): AN-hgdp-melanesian-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-papuan-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-papuan-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-biakapygmy-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-biakapygmy-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-san-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-san-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-mbutipygmy-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-mbutipygmy-XY-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-melanesian-XX-adj is not in table's info field
INFO (gnomad.assessment.sanity_checks 169): nhomalt-hgdp-melanesian-XY-adj is not in table's info field
```